CondTools/HLT/src/AlCaRecoTriggerBitsRcdUpdate.cc

```diff
@@ -78,8 +76,13 @@ void AlCaRecoTriggerBitsRcdUpdate::analyze(const edm::Event &evt, const edm::Eve
     return;
   }
 
-  // create what to write - starting from empty or existing list (auto_ptr?)
-  AlCaRecoTriggerBits *bitsToWrite = this->createStartTriggerBits(startEmpty_, iSetup);
+  // create what to write - starting from empty or existing list
+  std::unique_ptr<AlCaRecoTriggerBits> bitsToWrite;
+  if (startEmpty_) {
+    bitsToWrite = std::make_unique<AlCaRecoTriggerBits>();
+  } else {
+    bitsToWrite = std::make_unique<AlCaRecoTriggerBits>(iSetup.getData(triggerBitsToken_));
+  }
```

Comment:
I was reading [here (slide 5)](https://indico.cern.ch/event/1088598/contributions/4580152/attachments/2333275/3976721/DBOutputService%20changes.pdf) that stack should be preferred to heap, so what's behind the choice of `unique_ptr` here?

Reply:
This was motivated by code reduction, i.e. since we need to initialize the object in two different ways (empty or existing list) we either use the unique_ptr or we duplicate the code and create two objects, one of which will not be used. Hope this clarifies?

Reply:
Makes sense. Thanks.